src/test/kotlin/org/rust/ide/intentions/ToggleFeatureIntentionTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.IntentionActionDelegate
+import com.intellij.util.ui.UIUtil
+import org.intellij.lang.annotations.Language
+import org.rust.*
+import org.rust.cargo.RsWithToolchainTestBase
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.workspace.FeatureState
+import org.rust.cargo.project.workspace.PackageOrigin
+
+class ToggleFeatureIntentionTest : RsWithToolchainTestBase() {
```

Comment:
I recently added `@MockCargoFeatures`. It should be possible to write these tests without `RsWithToolchainTestBase` using it.

Reply:
`element.containingCargoPackage` is null in tests if I use only `@MockCargoFeatures`. Should I query the element's package in another way?

Reply:
I'm absolutely sure it shouldn't be null... Well, you can leave the tests as is, I'll try to convert them by myself

Reply:
Maybe I'm doing something wrong.

```kotlin
class ToggleFeatureIntentionTest : RsIntentionTestBase(ToggleFeatureIntention::class) {
    @MockCargoFeatures("foo")
    fun `test enable`() = doTest(
        """
        #[cfg(feature = "foo"/*caret*/)]
        fn foo() {}
    """, "foo", FeatureState.Disabled
    )

    private fun doTest(@Language("Rust") code: String, featureName: String, initialState: FeatureState) {
        val cargoProject = project.cargoProjects.singleProject()
        val pkg = cargoProject.workspaceOrFail().packages.single { it.origin == PackageOrigin.WORKSPACE }
        val feature = pkg.features.find { it.name == featureName } ?: error("Feature $featureName not found in test")
        project.cargoProjects.modifyFeatures(cargoProject, setOf(feature), initialState)

        InlineFile(code.trimIndent(), fileName).withCaret()
        launchAction()

        val cargoProjectRefreshed = project.cargoProjects.singleProject()
        val pkgRefreshed = cargoProjectRefreshed.workspaceOrFail().packages.single { it.origin == PackageOrigin.WORKSPACE }

        assertEquals(!initialState, pkgRefreshed.featureState[featureName])
    }
}
```

Reply:
Yep, the issue is in `fileName`. This argument should not be passed to `InlineFile` function. I.e. the file name must be "main.rs" (which is default argument value)

Reply:
Thank you! :)